python/ray/data/_internal/iterator/stream_split_iterator.py

```diff
@@ -235,6 +235,16 @@ def get(
             return block
         except StopIteration:
             return None
+        finally:
+            stats = (
+                self._executor.get_stats()
+                if self._executor
+                else self._base_dataset._plan.stats()
+            )
```

Comment:
maybe we can rewrite `SplitCoordinator.stats()` to return this instead of `DatasetStatsSummary`, and use the method here